subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -3981,6 +3981,9 @@ int lwm2m_engine_context_close(struct lwm2m_ctx *client_ctx)
 	struct observe_node *obs, *tmp;
 	sys_snode_t *prev_node = NULL;
 	int sock_fd = client_ctx->sock_fd;
+	struct lwm2m_message *msg;
+	size_t len = CONFIG_LWM2M_ENGINE_MAX_MESSAGES;
```

Comment:
Minor nit: we can drop the `len` variable here

and change the `for` loop that uses it to this below:
```
	for (i = 0, msg = messages; i < CONFIG_LWM2M_ENGINE_MAX_MESSAGES;
	     i++, msg++) {
```